module/zfs/dsl_scan.c

```diff
@@ -611,11 +612,13 @@ dsl_scan_prefetch(dsl_scan_t *scn, arc_buf_t *buf, blkptr_t *bp,
 	    (BP_GET_LEVEL(bp) == 0 && BP_GET_TYPE(bp) != DMU_OT_DNODE))
 		return;
 
+	if (BP_IS_ENCRYPTED(bp))
+		zio_flags |= ZIO_FLAG_RAW;
```

Comment:
Since this is the prefetch i/o, are you sure RAW would cause a problem?

The check on 612 still lets through non-dnode (and non-encrypted) blocks - specifically indirect blocks.  If it would be bad to use RAW on indirect blocks, we should have an assertion here.